### PR TITLE
DDA-228: Integrate diograph-js

### PR DESCRIPTION
**Pull request by jvalanen**
Diograph-js integration:
* Add diograph-js to diory-browser-electron as a submodule (to root)
* Load diograph-js in project package.json: `"diograph-js": "./diograph-js"`
* Verify integration by loading diory-demo-content diograph & getting generic-content diory 
* CI pipeline: Retrieve submodule and build diograph-js `dist/` folder as a part of CI pipeline

Ffmpeg initialization
* Add ffmpeg-static dependency to diory-browser-electron as diograph-js needs it
* Compose FFMPEG_PATH by using ffmpeg-static npm package (diograph-js video thumbnails require it)

CI & release scripts
* Pull and initialise diograph-js submodule in CI
* Clone with --recurse-submodules in release script + initialise diograph-js with yarn && yarn build
* Upgrade CI action versions
* NOTE: Uses always latest diograph-js commit (because don't know how to set environmental variable universally for both Mac & Windows)

Other 
* Define node version in `.npmrc`
* Use `electron-log` package to write all the console.log's in the backend to log files
  * Log files in `~/Library/Logs/Diory`
* Remove obsolete constants

### CODE REVIEW

File: electron/testing-diograph-js-integration.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { DiographJson } = require('../diograph-js')
```
Comment (opohjola) [inline electron/testing-diograph-js-integration.js:1]: Could we separate tests/specs clearly from the production code?
- separate tests -folder or similar
- unit specs using .spec -postfix

File: package-common.sh [hunk 2]
```diff
@@ -1,7 +1,17 @@
 # !/bin/bash
```
Comment (opohjola) [inline package-common.sh:1]: Could we move all packaging / publishing files to own folder?

File: electron-main.js [hunk 3]
```diff
@@ -5,6 +5,20 @@ const { default: installExtension, REDUX_DEVTOOLS } = require('electron-devtools
 
 require('electron-reload')
 
+// Electron logging
+// eslint-disable-next-line import/newline-after-import
+const log = require('electron-log')
+console.log = log.info
+
+// REMOVE ME: After first proper use of diograph-js this should be removed
+require('./electron/testing-diograph-js-integration')
```
Comment (opohjola) [inline electron-main.js:14]: Ok, this was a temporary thing.

**Comment by opohjola** [comment]
Ok, this was just about enabling diograph-js use.

**Review by opohjola — APPROVED** [review]

File: package-common.sh (hunk 2 above)
Comment (jvalanen) [reply package-common.sh:1]: I can try to make it better next time we do release. Haven't thought that it yet matters and haven't had good vision about it. But at least some of those which are currently in root may fit to scripts folder... 🤔

**Pull request merged**